src/utils/prepare-file.js

```diff
@@ -77,30 +84,79 @@ function prepareFile (file, opts) {
   let files = [].concat(file)
 
   return flatmap(files, (file) => {
-    if (typeof file === 'string') {
-      if (!isNode) {
-        throw new Error('Can only add file paths in node')
-      }
+    return prepare(file, opts)
+  })
+}
 
-      return loadPaths(opts, file)
-    }
+function prepare (file, opts) {
+  // probably it should be valid and would be handled below with Buffer.from
+  if (typeof file === 'string') {
+    throw new Error('String isn\'t valid as an input')
+  }
 
-    if (file.path && !file.content) {
-      file.dir = true
-      return file
-    }
+  // needs to test for stream because fs.createReadStream has path prop and would handle here
+  if (!isStream(file) && file.path && !file.content) {
+    file.dir = true
+    return file
+  }
 
-    if (file.content || file.dir) {
-      return file
+  if (file.content || file.dir) {
+    return file
+  }
+
+  if (isBrowser && file instanceof self.File) {
+    return {
+      path: file.name,
+      symlink: false,
+      dir: false,
+      content: fileReaderStream(file, opts)
     }
+  }
 
+  if (!isStream(file) && !isSource(file) && !Buffer.isBuffer(file)) {
     return {
       path: '',
       symlink: false,
       dir: false,
-      content: file
+      content: Buffer.from(file)
     }
-  })
+  }
+
+  return {
+    path: '',
+    symlink: false,
+    dir: false,
+    content: file
```

Comment:
I think it would be cleaner to have a single return. Then the above conditions would set only `path` and `content`. What do you think?